### DISCUSSION TITLE
Is only modifications to files prevented, can new files be created?

### DISCUSSION THREAD

**Discussion by CarrotManMatt** [xalvarez/prevent-file-change-action#841] [Q&A] [answered]
If I have the pattern: ".*\/MyProject\/myapp\/migrations\/\d{4}\w*\.py" to prevent modifications to any [Django migration files](https://docs.djangoproject.com/en/stable/topics/migrations/#migration-files), will this action also prevent new files from being created along with preventing the existing files in the `migrations/` directory from being edited?

I'd love to have a way to prevent changes to existing files, but allow new files to be added.

**Comment by xalvarez** [comment]
As of now the answer would be that the Action prevents adding new matching files. Allowing adding matching files is something I can see makes sense although perhaps I wouldn't see it as the default behaviour.

One way to implement this would be to add a new input, e.g. allowNewFiles, which defaults to false. Then, when calling octokit's listFiles method, if allowNewFiles is true, one could check the status property in the response and see if it has the value "added" for a matching file and if yes, allow this change.

Would such a solution match your expectations?

**Reply by CarrotManMatt** [reply]
Yeah that sounds awesome! If you need any help testing I'd be happy to, I don't have much experience with being able to help your development though.

**Reply by xalvarez** [reply]
No problem, I'll take care of it this week 🙂

**Reply by xalvarez** [reply, marked as answer]
Hi @CarrotManMatt, the changes are available under the tag v1.6.0, hope that helps.
The v1 tag has been updated to match v1.6.0.

**Reply by CarrotManMatt** [reply]
This is awesome! Thank you so much for implementing this so hastily.
